### PR TITLE
Feat/add dataverse filters

### PR DESCRIPTION
**Pull request by lolottetheclash**
This PR implements the filters according to this outcome: https://github.com/okp4/dataverse-portal/issues/88
here are the screenshots of the different screens. 

<details>
<summary>Large:</summary>

Capture d’écran 2023-03-23 à 11 32 44
Capture d’écran 2023-03-23 à 11 42 30
</details>

<details>
<summary>Medium:</summary>

Capture d’écran 2023-03-23 à 11 32 52
Capture d’écran 2023-03-23 à 11 42 36
</details>

<details>
<summary>Tablet:</summary>

Capture d’écran 2023-03-23 à 11 33 02

Capture d’écran 2023-03-23 à 11 33 09

Capture d’écran 2023-03-23 à 11 42 58

Capture d’écran 2023-03-23 à 11 43 04

</details>

<details>
<summary>Mobile:</summary>

Capture d’écran 2023-03-23 à 11 41 41

Capture d’écran 2023-03-23 à 11 35 22

Capture d’écran 2023-03-23 à 11 43 31

Capture d’écran 2023-03-23 à 11 43 36

</details>

### CODE REVIEW

File: src/page/dataverse/dataverse.tsx [hunk 1]
```diff
@@ -144,32 +180,166 @@ const renderTagColor = (itemType: DataverseItem): ColorVariant => {
...
+  const [dataverseResources, setDataverseResources] =
+    useState<DataverseItemDetails[]>(dataverseItems)
+
+  const isLargeScreen = isDesktop || isLargeDesktop
+
+  const filtersLabel = t('filters')
+
+  const renderLabel = useCallback((itemType: DataverseItem) => t(`resources.${itemType}`), [t])
+
+  const addAllFilter = useCallback((): void => {
```
Comment (ccamel) [inline src/page/dataverse/dataverse.tsx:219]: It's not an "addition", it more like "reset".

**Review by ccamel — CHANGES_REQUESTED** [review]
A few comments that can be discussed. ☺️

File: src/page/dataverse/dataverse.tsx [hunk 2]
```diff
@@ -144,32 +180,166 @@ const renderTagColor = (itemType: DataverseItem): ColorVariant => {
...
+  }, [])
+
+  const removeFilter = useCallback(
+    (filterToRemove: FilterValue): void => {
+      const isAllOnlyFilter = selectedFilters.length === 1 && selectedFilters.includes('all')
+      !isAllOnlyFilter &&
+        setSelectedFilters(selectedFilters.filter(filter => filter !== filterToRemove))
+    },
+    [selectedFilters]
+  )
```
[the remark this answers is not in the document]
Comment (lolottetheclash) [reply src/page/dataverse/dataverse.tsx:230]: 👍

File: src/component/button/button.tsx [hunk 3]
```diff
@@ -11,10 +12,11 @@ type ButtonProps = {
     startIcon?: JSX.Element
     endIcon?: JSX.Element
   }
-  variant?: 'primary' | 'secondary' | 'tertiary'
+  variant?: 'primary-blue' | 'primary-orange' | 'secondary' | 'tertiary'
```
[the remark this answers is not in the document]
Comment (AnjaMis) [reply src/component/button/button.tsx:15]: Maybe we can think of a better naming system for buttons, we have the same issues in another PR... At least in the mockup, all primary buttons are orange variants, and the secondary buttons are blue variants 🤔

**Review by fredericvilcot — APPROVED** [review]

**Pull request merged**